### PR TITLE
[libunwind] Add libunwind v1.7.2

### PR DESCRIPTION
**Pull request by ararslan**
This allows us to drop all but one patch.

### CODE REVIEW

**Comment by ararslan** [comment]
Hmm, well at least it builds successfully on FreeBSD.

**Comment by giordano** [comment]
> Hmm, well at least it builds successfully on FreeBSD.

You did it on purpose, didn't you?  Also, I thought we were using LLVMUnwind on FreeBSD?

**Comment by ararslan** [comment]
> You did it on purpose, didn't you?

😈

> I thought we were using LLVMUnwind on FreeBSD?

I was never able to get profiling to work with LLVMUnwind on FreeBSD, unfortunately.

**Comment by ararslan** [comment]
Still can't find `stdatomic.h` on Linux with `preferred_gcc_version=v"4.9"` even though it was definitely [added in 4.9](https://gcc.gnu.org/gcc-4.9/changes.html). 🤔

**Comment by giordano** [comment]
It works locally for me.  Great.

**Comment by giordano** [comment]
Ah, it worked for me because I used `host` as platform, which forced libgfortran5 and hence gcc 8, but if I compile for `x86_64-linux-gnu` I get
```console
sandbox:${WORKSPACE}/srcdir/libunwind-1.70 # cc --version
x86_64-linux-gnu-gcc (GCC) 4.8.5
Copyright (C) 2015 Free Software Foundation, Inc.
This is free software; see the source for copying conditions.  There is NO
warranty; not even for MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.
```
For some reasons I'm currently missing asking for 4.9 still gets you 4.8, this is unexpected :thinking:

**Comment by giordano** [comment]
```julia
julia> BinaryBuilderBase.select_closest_version(v"4.9", getfield.(BinaryBuilderBase.available_gcc_builds, :version))
v"4.8.5"
```
wut

**Comment by giordano** [comment]
```
[23:48:10] In file included from /opt/aarch64-linux-musl/aarch64-linux-musl/sys-root/usr/include/asm/ptrace.h:26:0,
[23:48:10]                  from ptrace/_UPT_reg_offset.c:32:
[23:48:10] /opt/aarch64-linux-musl/aarch64-linux-musl/sys-root/usr/include/asm/sigcontext.h:28:8: error: redefinition of ‘struct sigcontext’
[23:48:10]  struct sigcontext {
[23:48:10]         ^
[23:48:10] In file included from /opt/aarch64-linux-musl/aarch64-linux-musl/sys-root/usr/include/signal.h:48:0,
[23:48:10]                  from /opt/aarch64-linux-musl/aarch64-linux-musl/sys-root/usr/include/ucontext.h:9,
[23:48:10]                  from ../include/libunwind-aarch64.h:36,
[23:48:10]                  from ../include/libunwind.h:7,
[23:48:10]                  from ../include/libunwind-ptrace.h:29,
[23:48:10]                  from ptrace/_UPT_internal.h:44,
[23:48:10]                  from ptrace/_UPT_reg_offset.c:27:
[23:48:10] /opt/aarch64-linux-musl/aarch64-linux-musl/sys-root/usr/include/bits/signal.h:18:16: note: originally defined here
[23:48:10]  typedef struct sigcontext {
[23:48:10]                 ^
```
Sigh.

**Comment by ararslan** [comment]
> wut

That seems to be because the two oldest GCC versions we have listed in `available_gcc_builds` are 4.8.5 and 5.2.0, and `select_closest_version` picks 4.8.5 since it's closer to 4.9.

**Comment by giordano** [comment]
I find the definition of "closeness" in `select_closest_version` a bit questionable...

**Comment by ararslan** [comment]
Everyone knows that the optimal version number is found by minimizing the $\ell_1$ norm of the component-wise differences!

I wonder if `preferred_gcc_version` should actually be something like `minimum_gcc_version`, which could then use `searchsortedfirst` to get oldest version that meets or exceeds the minimum.

**Comment by ararslan** [comment]
Regarding the musl failure, I wonder if we could (conditionally?) override `HAVE_ASM_PTRACE_H`, which is the guard for the problematic include.

**Comment by giordano** [comment]
```
fatal error: stdatomic.h: No such file or directory
```
maybe just requires a newer GCC?

File: L/LibUnwind/LibUnwind@1.7.2/build_tarballs.jl [hunk 1]
```diff
@@ -0,0 +1,67 @@
...
+    BuildDependency("XZ_jll"),
+    Dependency("Zlib_jll"),
+    BuildDependency("LLVMCompilerRT_jll"; platforms=[Platform("x86_64", "linux"; sanitize="memory")]),
+]
+
+# Build the tarballs. Note that libunwind started using `stdatomic.h`, which is only
+# available with GCC version 4.9 or later, so we need to set a higher preferred version
+# than the default.
+build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
+               julia_compat="1.10", preferred_gcc_version=v"4.9")
```
Comment (giordano) [inline L/LibUnwind/LibUnwind@1.7.2/build_tarballs.jl:67]:
```suggestion
# Build the tarballs. Note that libunwind started using `stdatomic.h`, which is only
# available with GCC version 4.9 or later, so we need to set a higher preferred version
# than the default.
build_tarballs(ARGS, name, version, sources, script, platforms, products, dependencies;
               julia_compat="1.10", preferred_gcc_version=v"5")
```

Comment (maleadt) [reply L/LibUnwind/LibUnwind@1.7.2/build_tarballs.jl]: That's what I did in the other PR. Is there a reason we wouldn't want this?

**Comment by giordano** [comment]
Ok, now we're left only with the aarch64-linux-musl error [shown above](https://github.com/JuliaPackaging/Yggdrasil/pull/6843#issuecomment-1585290286).  Suggestions for what to do with that are welcome

**Comment by maleadt** [comment]
What about patching the system? Nasty, I know, but :shrug:

**Pull request merged**